proposals/nnnn-screen-manager-alerts.md

```diff
@@ -0,0 +1,231 @@
+
+/**
+ The underlying prompt generated by the `init`.
+ */
+@property (nullable, copy, nonatomic, readonly) NSArray<SDLTTSChunk *> *prompt;
+
+/**
+ If created with an audio file, the audio file that will be uploaded and used.
+ */
+@property (nullable, copy, nonatomic, readonly) SDLFile *audioFile;
+
+/**
+ Initialize with an SDLFile holding data or pointing to a file on the file system. When this object is passed to an `Alert` or `Speak`, the file will be uploaded if it is not already, then played if the system supports that feature.
+
+ Only available on SDL Core 5.0+.
+
+ @param audioFile The audio file to be played by the system
+ @param tone Whether or not to play a system tone before the audio file
+ */
+- (instancetype)initWithAudioFile:(SDLFile *)audioFile playSystemTone:(BOOL)tone;
+
+/**
+ Initialize with a string to be spoken by the system speech synthesizer.
+
+ @param spokenString The string to be spoken by the system speech synthesizer
+ @param tone Whether or not to play a system tone before the synthesized speech
+ */
+- (instancetype)initWithSpeechSynthesizerString:(NSString *)spokenString playSystemTone:(BOOL)tone;
+
+/**
+ Initialize with a string to be spoken by the system speech synthesizer using a phonetic string.
+
+ @param spokenString The string to be spoken by the system speech synthesizer
+ @param phoneticType Must be one of `SAPI_PHONEMES`, `LHPLUS_PHONEMES`, `TEXT`, or `PRE_RECORDED` or no object will be created
+ @param tone Whether or not to play a system tone before the synthesized speech
+ */
+- (instancetype)initWithPhoneticSpeechSynthesizerString:(NSString *)phoneticString phoneticType:(SDLSpeechCapabilities)phoneticType playSystemTone:(BOOL)tone;
+
+/**
+ Initialize with only a tone.
+ */
+- (instancetype)initWithTone;
+```
+
+##### Java
+```java
+public class AlertAudioData {
+    private Bool playTone;
+    private List<TTSChunk> prompt;
+    private SdlFile audioFile;
+
+    // All vars have getters as well but no setters
+
+    AlertAudioData(@NonNull SdlFile audioFile, @NonNull Bool playTone)
+    AlertAudioData(@NonNull String spokenString, @NonNull Bool playTone)
+    AlertAudioData(@NonNull String phoneticString, @NonNull SpeechCapabilities phoneticType, @NonNull Bool playTone)
+    AlertAudioData(@NonNull Bool playTone)
+}
+```
+
+### Manager Alert API
+The next object is the alert view itself that developers will construct and pass to the Screen Manager.
+
+##### iOS
+```objc
+@interface SDLAlertView: NSObject
+
+/**
+ Set this to change the default timeout for all alerts. If a timeout is not set on an individual alert object (or if it is set to 0.0), then it will use this timeout instead. See `timeout` for more details. If this is not set by you, it will default to 5 seconds. The minimum is 3 seconds, the max is 10 seconds. If this is set below the minimum, it will be capped at 3 seconds. If this is set above the maximum, it will be capped at 10 seconds.
```

Comment:
```suggestion
 Set this to change the default timeout for all alerts. If a timeout is not set on an individual alert object (or if it is set to 0.0), then it will use this timeout instead. See `timeout` for more details. If this is not set by you, it will default to 5 seconds. The minimum is 3 seconds, the maximum is 10 seconds. If this is set below the minimum, it will be capped at 3 seconds. If this is set above the maximum, it will be capped at 10 seconds.
```